File: Python/0011. Olympic Game.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    
    while True:
        try:
            m = int(input())
            arr = []
            for _ in range(m):
                arr.append(list(input().strip().split()))
            arr.sort(key = lambda x:(-int(x[1]),-int(x[2]),-int(x[3]),x[0]))
            for x in arr:
                print(x[0])
            return
        
        except EOFError:
            break

    
if __name__ == "__main__":
    func()

File: C++/0011. Olympic Game.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

#include<tuple>
#include<vector>
#include<algorithm>
bool compare_tuple(const tuple<string,int,int,int>&x, const tuple<string,int,int,int>&y){
    if (get<1>(x) == get<1>(y)){
        if (get<2>(x) == get<2>(y)){
            if (get<3>(x) == get<3>(y)){
                return get<0>(x) < get<0>(y);
            }
            else{
                return get<3>(x) > get<3>(y);
            }
        }
        else{
            return get<2>(x) > get<2>(y);
        }
    }   
    else{
        return get<1>(x) > get<1>(y);
    }
}

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    int n;
    while(cin>>n){
        string country;
        int gold;
        int silver;
        int bronze;
        vector<tuple<string,int,int,int>> arr(n);
        for(int i=0;i<n;++i){
            cin>>country;
            cin>>gold;
            cin>>silver;
            cin>>bronze;
            arr[i] = make_tuple(country,gold,silver,bronze);
        }
        sort(arr.begin(),arr.end(),compare_tuple);
        
        for(auto x:arr){
            cout<<get<0>(x)<<endl;
        }
        
    }
    return 0;
}
